ibridges/session.py

```diff
@@ -106,8 +108,36 @@ def has_valid_irods_session(self) -> bool:
         """
         return self.irods_session is not None and self.server_version != ()
 
+    def _network_check(self, hostname: str, port: int) -> bool:
+        """Check connectivity to an iRODS server.
+
+        Parameters
+        ----------
+        hostname : str
+            FQDN/IP of an iRODS server.
+
+        Returns
+        -------
+        bool
+            Connection to `hostname` possible.
+
+        """
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            try:
+                sock.settimeout(10.0)
+                sock.connect((hostname, port))
+                return True
+            except socket.error:
+                return False
+
+
     def connect(self) -> iRODSSession:
         """Establish an iRODS session."""
+        irods_host = self._irods_env.get('irods_host', '')
+        irods_port = int(self._irods_env.get('irods_port', ''))
+        network = self._network_check(irods_host, irods_port)
+        if network is False:
+            raise socket.error(f'No internet connection to {irods_host} and port {irods_port}')
```

Comment:
socket.error seems to be deprecated according to the documentation. What about a NetworkError?

Reply:
There is no generic python error called `NetworkError`. We have an irods.exception.NetworkError.
I can reuse this one.

Reply:
I found a `ConnectionError`; its is in the basic python exceptions. I think this one fits.